src/mempool_args.cpp

```diff
@@ -33,5 +33,12 @@ void ApplyArgsManOptions(const ArgsManager& argsman, MemPoolOptions& mempool_opt
 
     if (auto hours = argsman.GetIntArg("-mempoolexpiry")) mempool_opts.expiry = std::chrono::hours{*hours};
 
+
+    if (argsman.GetBoolArg("-persistmempool", DEFAULT_PERSIST_MEMPOOL)) {
+        if (mempool_opts.persistence_path.empty()) mempool_opts.persistence_path = argsman.GetDataDirNet() / "mempool.dat";
```

Comment:
Mostly because the other options specifies a value that goes into `mempool_opts`, whereas `-persistmempool` is a boolean option.

Note that we use an empty `persistence_path` to indicate that we don't want to persist the mempool

So the logic goes:

- If we want to persist the mempool, and there's no existing mempool path set, set `persistence_path` to `<datadir>/ mempool.dat`
- If we don't want to persist the mempool, set `persistence_path` to an empty path